src/libs/TransactionUtils.ts

```diff
@@ -693,6 +704,10 @@ function hasNoticeTypeViolation(transactionID: string, transactionViolations: On
  * Checks if any violations for the provided transaction are of type 'warning'
  */
 function hasWarningTypeViolation(transactionID: string, transactionViolations: OnyxCollection<TransactionViolation[]>): boolean {
+    if (!Permissions.canUseDupeDetection(allBetas ?? [])) {
```

Comment:
It caused [this bug](https://github.com/Expensify/App/issues/50799). With this change, hasWarningTypeViolation always returns false if canUseDupeDetection is false, but dupe detection is only one reason that causes a warning violation type. When a receipt scanning fails, the BE still returns a warning violation